Removed different types of Detectors for Resources. (#1810)
open-telemetry/opentelemetry-go: CHANGELOG.md

### Added

> 

Reviewer: ```suggestion
```

### Changed

- Make `NewSplitDriver` from `go.opentelemetry.io/otel/exporters/otlp` take variadic arguments instead of a `SplitConfig` item.
  `NewSplitDriver` now automically implements an internal `noopDriver` for `SplitConfig` fields that are not initialized. (#1798)
- `resource.New()` now creates a Resource without builtin detectors. Previous behavior is now achieved by using `WithBuiltinDetectors` Option. (#1810)

### Deprecated

### Removed

> - Remove `resource.WithoutBuiltin()`. Use `resource.New()`. (#1810)
> - Unexported types `resource.FromEnv`, `resource.Host`, and `resource.TelemetrySDK`, Use the corresponding `With*()` to use individually. (#1810)
> ### Fixed

Reviewer: ```suggestion

- Remove `resource.WithoutBuiltin()`. Use `resource.New()`. (#1810)
- Unexported types `resource.FromEnv`, `resource.Host`, and `resource.TelemetrySDK`, Use the corresponding `With*()` to use individually. (#1810)

### Fixed
```

- The `Event` and `Link` struct types from the `go.opentelemetry.io/otel` package now include a `DroppedAttributeCount` field to record the number of attributes that were not recorded due to configured limits being reached. (#1771)
- The Jaeger exporter now reports dropped attributes for a Span event in the exported log. (#1771)
- Adds `k8s.node.name` and `k8s.node.uid` attribute keys to the `semconv` package. (#1789)

> - Adds `resource.NewEmptyResouce()` for creating a Resouce without builtin detectors (#1810)

Reviewer 2: ```suggestion
- Adds `resource.NewEmptyResource()` for creating a Resource without builtin detectors (#1810)
```

`NewEmptyResource` instead of `NewEmptyResouce`?

> - Adds `resource.NewEmptyResource()` for creating a Resource without builtin detectors (#1810)

Reviewer: ```suggestion
- Adds `resource.NewEmptyResource()` for creating a Resource without builtin detectors. (#1810)
```

- The `trace.FlagsDebug` and `trace.FlagsDeferred` constants have been removed and will be localized to the B3 propagator. (#1770)
- Remove `Process` configuration, `WithProcessFromEnv` and `ProcessFromEnv`, from the Jaeger exporter package.
  The information that could be configured in the `Process` struct should be configured in a `Resource` instead. (#1776)

> - Remove `resource.WithoutBuiltin()`.  Any use of `resource.WithDetecors()` will have the same effect (#1810)

Reviewer 3: ```suggestion
- Remove `resource.WithoutBuiltin()`.  Any use of `resource.WithDetectors()` will have the same effect (#1810)
```

  The information that could be configured in the `Process` struct should be configured in a `Resource` instead. (#1776, #1804)
- Remove the `WithDisabled` option from the Jaeger exporter.
  To disable the exporter unregister it from the `TracerProvider` or use a no-operation `TracerProvider`. (#1806)
- Remove `resource.WithoutBuiltin()`. Use `resource.NewEmptyResource()`. (#1810)

> - Unexported types `resouce.FromEnv`, `resource.Host`, and `resource.TelemetrySDK`, Use the corresponding `With*()` to use individually. (#1810)

Reviewer 2: ```suggestion
- Unexported types `resource.FromEnv`, `resource.Host`, and `resource.TelemetrySDK`, Use the corresponding `With*()` to use individually. (#1810)
```
